src/signTx.c

```diff
@@ -806,8 +806,8 @@ static void parseInput(uint8_t* wireDataBuffer, size_t wireDataSize)
 
 	VALIDATE(wireDataSize == SIZEOF(*wireUtxo), ERR_INVALID_DATA);
 
-	memmove(input->txHashBuffer, wireUtxo->txHash, SIZEOF(input->txHashBuffer));
-	input->parsedIndex = u4be_read(wireUtxo->index);
+	memmove(input->input_data.txHashBuffer, wireUtxo->txHash, SIZEOF(input->input_data.txHashBuffer));
```

Comment:
Instead of `sign_tx_transaction_input_t* input`, let's make a pointer to input->input_data at the beginning of this function.

Reply:
We can also move the pointer declaration closer to its use (after VALIDATE).